sky/data/storage.py

```diff
@@ -1865,10 +1902,615 @@ def _delete_gcs_bucket(self, bucket_name: str) -> bool:
+                            'container under an already existing storage '
+                            'account within your subscription ID.'
+                        )
+            else:
+                # If storage account name is not provided from config, then 
+                # use default resource group and storage account names.
+                self.storage_account_name = f'sky{self.region}{common_utils.get_user_hash()}'
+                self.resource_group_name = f'sky{common_utils.get_user_hash()}'
+                self._init_storage_account_and_resource_group()
+
+        # resource_group_name is set to None when using non-sky-managed
+        # public container or private container without authorization.
+        if self.resource_group_name is not None:
+            self.storage_account_key = data_utils.get_az_storage_account_key(
+                self.storage_account_name, self.resource_group_name,
+                self.storage_client, self.resource_client)
+
+        self.bucket_name, is_new_bucket = self._get_bucket()
+        if self.is_sky_managed is None:
+            # If is_sky_managed is not specified, then this is a new storage
+            # object (i.e., did not exist in global_user_state) and we should
+            # set the is_sky_managed property.
+            # If is_sky_managed is specified, then we take no action.
+            self.is_sky_managed = is_new_bucket
+
+    def _init_storage_account_and_resource_group(self) -> None:
+        """Initialize storage account and resource group.
+
+        Checks if the set resource group name and storage account name already
+        exists under user's subscription ID. If not, creates them and assigns
+        Storage Blob Data Owner role of the storage account.
+
+        Raises:
+            StorageBucketCreateError: If storage account attempted to be
+                created already exists
+        """
+        try:
+            # obtains detailed information about resource group under the
+            # user's subscription. Used to check if the name already exists
+            self.resource_client.resource_groups.get(
+                self.resource_group_name)
+        except azure.exceptions().ResourceNotFoundError as e:
+            if 'Code: ResourceGroupNotFound' in e.message:
+                with rich_utils.safe_status(
+                    f'[bold cyan]Setting up:\n'
+                    f'- Resource group: {self.resource_group_name}'):
+                    self.resource_client.resource_groups.create_or_update(
+                        self.resource_group_name, {'location': self.region})
+        # check if the storage account name already exists under the
+        # given resource group name.
+        try:
+            self.storage_client.storage_accounts.get_properties(
+                self.resource_group_name, self.storage_account_name)
+        except azure.exceptions().ResourceNotFoundError as e:
+            if 'Code: ResourceNotFound' in e.message:
```

Comment:
same here, what if we enter the else branch

Reply:
Fixed. Updated at [3e8c96e](https://github.com/skypilot-org/skypilot/pull/3032/commits/3e8c96e42689ba2b5f75ed570c6d3d334914aa87)

Reply:
Done https://github.com/skypilot-org/skypilot/pull/3032#discussion_r1646594533